src/main/java/io/vlingo/schemata/codegen/TypeDefinitionCompilerActor.java

```diff
@@ -33,12 +38,18 @@ public TypeDefinitionCompilerActor(final TypeParser parser, final List<Processor
     }
 
     @Override
-    public Completes<String> compile(final InputStream typeDefinition, final String fullyQualifiedTypeName, final String version) {
+    public Completes<Outcome<SchemataBusinessException, String>> compile(final InputStream typeDefinition, final String fullyQualifiedTypeName, final String version) {
         CompletesEventually eventually = completesEventually();
 
         parser.parseTypeDefinition(typeDefinition, fullyQualifiedTypeName)
-                .andThenTo(this.process(fullyQualifiedTypeName))
-                .andThenTo(tree -> backend.generateOutput(tree, version))
+                .andThen(o -> o.resolve(
+                        ex -> Failure.of(SchemataBusinessException.invalidSchemaDefinition(ex)),
+                        node -> Success.of(this.process(fullyQualifiedTypeName).apply(node))
+                ))
+                .andThen(o -> o.resolve(
+                        Failure::of,
+                        node -> Success.of(backend.generateOutput((Node) node, version))
+                ))
                 .andThenConsume(eventually::with);
```

Comment:
`process#apply` returns as `Complets<Node>`, not a `Node`. So you end up with `Completes<Outcome<Error, Completes<Node>>`. How do we turn that into a `Completes<Outcome<Error, Node>>`?

Reply:
It's the flatMap from hell. (tm)